wire-ios-data-model/Source/Model/FeatureConfig/Feature.swift

```diff
@@ -136,9 +136,11 @@ public class Feature: ZMManagedObject {
         fetchRequest.predicate = NSPredicate(format: "nameValue == %@", name.rawValue)
         fetchRequest.fetchLimit = 2
 
-        let results = context.fetchOrAssert(request: fetchRequest)
-        require(results.count <= 1, "More than instance for feature: \(name.rawValue)")
-        return results.first
+        return context.performAndWait {
+            let results = context.fetchOrAssert(request: fetchRequest)
+            require(results.count <= 1, "More than instance for feature: \(name.rawValue)")
+            return results.first
+        }
```

Comment:
is this the right place to do it, or fetching the feature are supposed to be run from syncContext?

Reply:
I moved it to another place.